src/components/ListLayout.jsx

```diff
@@ -27,32 +30,87 @@ const ListLayout = ({ items, localToken }) => {
   useEffect(() => {
     inputRef.current.focus();
   }, []);
+  // updates isActive property of item to true if item has 2+ purchases and has been purchased within calculated estimate
+  // isActive is defaulted to false when item is added
+  useEffect(() => {
+    if (!loading) {
+      items.forEach((item) => {
+        //if item was bought within 24 hours gap it should be checked
+        /*   item.checked = within24hours(item.purchasedDate); */
+
+        const dateOfLastTransaction =
+          item.totalPurchases > 0 ? item.purchasedDate : item.createdAt;
+        const daysSinceLastTransaction =
+          (currentTime - dateOfLastTransaction) / oneDay;
+        if (
+          item.totalPurchases > 1 &&
+          daysSinceLastTransaction < 2 * item.previousEstimate
+        ) {
+          let itemToUpdate = {
+            isActive: true,
+          };
+          setUpdateToDb(localToken, item.id, itemToUpdate);
+        }
+      });
+
+      let newList = [];
+      items.forEach((item) => {
+        newList.push({ ...item, checked: within24hours(item.purchasedDate) });
+      });
+      setLayoutItems(newList);
+    }
+    //suggested dependency array via React and I agree with the suggestion if anyone has thoughts on this please let me know!
+  }, [items]);
```

Comment:
It looks like there need to be more dependencies in here (here's the error causing the github action to fail):
` Line 63:6:  React Hook useEffect has missing dependencies: 'currentTime', 'loading', 'localToken', and 'within24hours'. Either include them or remove the dependency array  react-hooks/exhaustive-deps`

Reply:
> Nice work, Alena! The checkbox functionality works well. I feel like adding a couple of comments to the changes you made to the `handleCheckBoxChange` would help clear up what is happening there.
yes, thanks, remember about comments just taking a break and look through the code again and add some comments

Reply:
> Maybe we should add a success toast message? That could be a TODO for design stuff too...

Great idea, we should